### ISSUE TITLE
Add twisted.web2.xmlrpc docs and example code.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11272]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)| @oubiwann reported|
|-|-|
|Trac ID|trac#1549|
|Type|task|
|Created|2006-03-06 18:41:21Z|
|Branch|https://github.com/twisted/twisted/tree/xmlrpc-docs-1549|

Itamar's XML-RPC code is going into web2 any day now, and it will need some docs. I've started putting some together, and when I'm done, will have them reviewed for inclusion.

Attachments:

* [xmlrpc.html](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/41e/41e842e76e8ea73f910aac95ea1ac9f0707643b2/9259683e0dc0de03dd90489aa477dda37603709c.html) (45441 bytes) - added by oubiwann on 2006-07-07 03:36:21Z - First draft of t.w2.xmlrpc howto

<details><summary>Searchable metadata</summary>

```
trac-id__1549 1549
type__task task
reporter__oubiwann oubiwann
priority__normal normal
milestone__Web2_Gold_Master Web2-Gold-Master
branch__branches_xmlrpc_docs_1549 branches/xmlrpc-docs-1549
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__web2 web2
keywords__xmlrpc_docs_demos_examlpes xmlrpc docs demos examlpes
time__1141670481000000 1141670481000000
changetime__1299821293000000 1299821293000000
version__ 
owner__ 
cc__itamarst cc__jknight cc__oubiwann cc__dreid cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

I hope you're basing them on the twisted.web xmlrpc docs? Less work.

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)|@oubiwann set status to `assigned`|
|-|-|

Yes, I've copied them over and have started updating them :-)

As for the format: James, is there an automated process that will get buggered if I keep the old format of xhtml? Are the ReST docs going away? Or should I convert the xhtml to ReST?

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)|@oubiwann commented|
|-|-|

I finally got a chance to start this tonight (at hacking society) and it's going smoothly. I have some suggestions, though:

1) ditch the .rpy listing

2) replace it with a similar example, showing the Quoter as a resource in a web2 server example -- putChild('RPC2', Quoter())

3) ditch the SOAP example listing

4) ditch the SOAP + XML-RPC example (as the new example above will show this basic functionality)

Later (when http-auth-1475 gets merged):

5) write a cred/auth example

Thoughts?

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

Sounds good to me.

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)|@oubiwann commented|
|-|-|

Just talked with James on IRC about formatting, etc.:

1) I'm going to convert to ReST

2) All the inline code will be moved into "examples"

3) this file will be named xmlrpc-howto.txt

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

ReST is not well supported by our current website integration infrastrcture; notice how web2 documentation link on site is broken...

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)|@oubiwann commented|
|-|-|

Here's a related ticket, regarding Itamar's last comment:

[#7111](https://github.com/twisted/twisted/issues/7111) - Web2 documentation should not disregard the Documentation Standard

In the comment from 03/15/06 15:51:56, dreid outlines a process, but on IRC he just said that this has not been decided as yet.

Regardless, I will continue work on the howto. For now, I will keep it in xhtml since it will be easier to convert from that to ReST than from ReST back to xhtml. I will wait on the patch until the doc format has reached some sort of conclusion.

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)|@oubiwann commented|
|-|-|

Ticket for the http auth branch: [#7128](https://github.com/twisted/twisted/issues/7128)

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)|@oubiwann commented|
|-|-|

I have the howto updated for twisted.web2 now, with the exception of a client (Proxy) example and an authentication example. I have opend ticket [[#1895](https://github.com/twisted/twisted/issues/1895)](https://github.com/twisted/twisted/issues/1895) for writing the client code. Unless asked to do otherwise, I will submit the docs for [#11272](https://github.com/twisted/twisted/issues/11272) when #1895 is closed and the Proxy patch is in trunk.

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)|@oubiwann commented|
|-|-|

I've added a first draft of the docs, so that the content can be reviewed. I've attached the HTML as generated by admin/process-docs, since the listings can be viewed easily with a single upload.

The only section I haven't worked on yet is the client section. It should be essentially identical to the old section, but for the sake of completeness, I am waiting to resolve the issues I am having with the new proxy code ([#1895](https://github.com/twisted/twisted/issues/1895)) first.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

The attached file is lore _output_.  Is it too much to hope that you still have the lore _input_ from which this was generated?

Whatever the case, we should adjust the document to cover `twisted.web.xmlrpc` instead.

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)<a name="note_14"></a>|@oubiwann commented|
|-|-|

I do believe it still exists on one of my laptops... yes, I have just confirmed that I kept it in an old checkout of Twisted (r17451):-)

I will update this ticket with a branch that contains the files (and then later update those files to use web instead of web2).

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)<a name="note_15"></a>|@oubiwann commented|
|-|-|

(In [24911]) Branching to 'xmlrpc-docs-1549'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_16"></a>|@thijstriemstra commented|
|-|-|

Can this be moved to web instead, since web2 is being deprecated, or should this ticket simply be closed?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun commented|
|-|-|

Twisted Web has some XML-RPC docs already.  Maybe someone could take a look at the branch here and decide if any of the docs represent improvements which could be ported back to Twisted Web.

I don't think there's a whole lot of value in finishing the web2 docs, but I'm not going to stop anyone from doing it either.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)|@washort set status to `closed`|
|-|-|
